### PR TITLE
Add doxygen GitHub action

### PR DESCRIPTION
**Pull request by slfuchs**

### CODE REVIEW

File: .github/workflows/doxygen.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      
+      - name: Compile
+        run: |
+          mkdir build
+          cd build
+          cmake ../ -DCMAKE_BUILD_TYPE=Release
+          make doxygen
+
+#      - name: Deploy
+#        uses: peaceiris/actions-gh-pages@v3
```
Comment (peterrum) [inline .github/workflows/doxygen.yml]:
```suggestion
#      - name: Deploy
#        if: ${{ ... }}
#        uses: peaceiris/actions-gh-pages@v3
```
You will need a check that the deployment only happens during push onto master.

File: .github/workflows/doxygen.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: Doxygen
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  schedule:
+  - cron:  '0 0 * * 0'
```
Comment (peterrum) [inline .github/workflows/doxygen.yml:9]:
```suggestion
  schedule:
  - cron:  '0 0 * * 0'
```

**Review by ppraegla — APPROVED** [review]

File: .github/workflows/doxygen.yml (hunk 2 above)
Comment (ppraegla) [reply .github/workflows/doxygen.yml:9]: I see no difference for your suggested change

**Comment by slfuchs** [comment]
This GitHub action generates the Doxygen documentation, checks for errors, and deploys the Doxygen documentation to [PartExa/PartExa-Doxygen](https://github.com/PartExa/PartExa-Doxygen) (with GitHub page https://partexa.github.io/PartExa-Doxygen/). Note that the `deploy` step is only executed for `push` and `schedule` events.

**Comment by slfuchs** [comment]
Access to the stored action deploy key is only possible for `push` and `schedule` events. For this reason, I guess we have to merge and see if the deploy stage works as expected.

**Pull request merged**